app.vue

```diff
@@ -216,12 +216,29 @@ useHead({
       return bodyClass.join(' ')
     }),
   },
-  script: [
-    {
-      innerHTML: `if('serviceWorker' in navigator){window.addEventListener('load', () => {navigator.serviceWorker.register('/sw.js', { scope: '/' })})}`,
-    },
-  ],
 })
+const hasUpgrade = ref<boolean>(false)
+if ('serviceWorker' in navigator) {
+  navigator.serviceWorker.register('/sw.js').then(registration => {
+    // Listen for updates
+    registration.addEventListener('updatefound', () => {
+      const newWorker = registration.installing
+      if (newWorker) {
+        newWorker.addEventListener('statechange', () => {
+          if (newWorker.state === 'activated') {
+            // Notify the user that an update is available
+            console.log('A new version is available. Please refresh.')
```

Comment:
from what I can see this is just logging on the console log and not really notifying the user. If he doesnt open the devtools he wouldn't know that he needs to refresh the page to get the update right?

Reply:
That was for debugging. Removing.